Description: foomatic-rip: Changed Ghostscript call to count pages in a PDF file to
 use "runpdfbegin" and not the undocumented Ghostscript internal "pdfdict".
Origin: vendor
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/cups-filters/+bug/1828401
Bug-Debian: https://bugs.debian.org/928936
Bug-Debian: https://bugs.debian.org/926576
Bug-ArchLinux: https://bugs.archlinux.org/task/62251
Forwarded: no
Author: Till Kamppeter <till.kamppeter@gmail.com>
Last-Update: 2019-05-13

---
 filter/foomatic-rip/pdf.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/filter/foomatic-rip/pdf.c b/filter/foomatic-rip/pdf.c
index 9c3979bb28bb..bc739bd80874 100644
--- a/filter/foomatic-rip/pdf.c
+++ b/filter/foomatic-rip/pdf.c
@@ -47,9 +47,8 @@ static int pdf_count_pages(const char *filename)
     size_t bytes;
 
     snprintf(gscommand, CMDLINE_MAX, "%s -dNODISPLAY -q -c "
-	     "'/pdffile (%s) (r) file def pdfdict begin pdffile pdfopen begin "
-	     "(PageCount: ) print pdfpagecount == flush currentdict pdfclose "
-	     "end end quit'",
+	     "'/pdffile (%s) (r) file runpdfbegin (PageCount: ) print "
+	     "pdfpagecount = quit'",
 	     gspath, filename);
 
     FILE *pd = popen(gscommand, "r");
-- 
2.20.1

